cluster/kubernetes/resource/service.go

```diff
@@ -12,12 +17,31 @@ type Service struct {
 	Spec ServiceSpec `yaml:"spec"`
 }
 
+func (o Service) ServiceIDs(all map[string]resource.Resource) []flux.ServiceID {
+	// A service is part of its own service id
+	ns := o.Meta.Namespace
+	if ns == "" {
+		ns = "default"
+	}
+	return []flux.ServiceID{flux.ServiceID(fmt.Sprintf("%s/%s", ns, o.Meta.Name))}
+}
+
+// Matches checks if this service's label selectors match the labels fo some
+// deployment
+func (o Service) Matches(l labels.Labels) bool {
+	return o.Spec.Matches(l)
+}
+
 type ServiceSpec struct {
 	Type     string            `yaml:"type"`
 	Ports    []ServicePort     `yaml:"ports"`
 	Selector map[string]string `yaml:"selector"`
 }
 
+func (s ServiceSpec) Matches(l labels.Labels) bool {
+	return labels.SelectorFromSet(labels.Set(s.Selector)).Matches(l)
```

Comment:
Unknown, there's also the `omitempty` issue with yaml parsing which will play into this.